lambda_functions/analyzer/binary_info.py

```diff
@@ -97,23 +97,24 @@ def save_matches_and_alert(self, lambda_version, dynamo_table_name, sns_topic_ar
 
     def summary(self):
         """Generate a summary dictionary of binary attributes."""
-        return {
+        yara_results = {
             'FileInfo': {
                 'ComputedMD5': self.computed_md5,
                 'ComputedSHA256': self.computed_sha,
                 'ReportedMD5': self.reported_md5,
                 'S3Location': self.s3_identifier,
                 'SamplePath': self.observed_path
             },
-            'MatchedRules': [
-                {
-                    # YARA string IDs, e.g. "$string1"
-                    'MatchedStrings': list(sorted(set(t[1] for t in match.strings))),
-                    'Meta': match.meta,
-                    'RuleFile': match.namespace,
-                    'RuleName': match.rule,
-                    'RuleTags': match.tags
-                }
-                for match in self.yara_matches
-            ]
+            'NumMatchedRules': len(self.yara_matches)
         }
+
+        for index, match in enumerate(self.yara_matches):
+            yara_results['MatchedRule{:03d}'.format(index+1)] = {
```

Comment:
It will actually be better to just have the raw number (not formatting with 3 zeros). There could in theory be any number of rules; simple consistent formatting will be easiest here. In addition, instead of `index+1`, you can use the `start` keyword:

```python
for index, match in enumerate(self.yara_matches, start=1):
    yara_results['MatchedRule{}'.format(index)] = {
```